src/api/app/models/token/workflow.rb

```diff
@@ -10,23 +10,35 @@ def call(options)
     return unless extractor.allowed_event_and_action?
 
     scm_extractor_payload = extractor.call
-    yaml_file = Workflows::YAMLDownloader.new(scm_extractor_payload).call
+    yaml_downloader = Workflows::YAMLDownloader.new(scm_extractor_payload)
+    yaml_file = yaml_downloader.call
+    if yaml_file.blank?
+      raise Token::Errors::NonExistentWorkflowsFile,
+            ".obs/workflows.yml could not be downloaded on the SCM branch #{scm_extractor_payload[:target_branch]}: #{yaml_downloader.errors.join('\n')}"
+    end
     workflows = Workflows::YAMLToWorkflowsService.new(yaml_file: yaml_file, scm_extractor_payload: scm_extractor_payload, token: self).call
+    invalid_workflow_error! if invalid_workflows?(workflows)
 
     workflows.each do |workflow|
       workflow.steps.each do |step|
         run_step_and_report(step, scm_extractor_payload, scm_token)
-      rescue Octokit::Unauthorized, Gitlab::Error::Unauthorized => e
-        raise Token::Errors::SCMTokenInvalid, e.message
       end
     end
+  rescue Octokit::Unauthorized, Gitlab::Error::Unauthorized => e
+    raise Token::Errors::SCMTokenInvalid, e.message
   end
 
   private
 
-  def run_step_and_report(step, scm_extractor_payload, scm_token)
-    raise 'Invalid workflow step definition' unless step.valid?
+  def invalid_workflow_error!
```

Comment:
Why not use the content of this method directly? Having a method for a one liner doesn't improve the code.

Same thing for `invalid_workflows?`.